sqlite/src/vdbeInt.h

```diff
@@ -796,6 +798,6 @@ int sqlite3LockStmtTables(sqlite3_stmt *);
 
 Mem* sqlite3GetCachedResultRow(sqlite3_stmt *pStmt, int *nColumns);
 
-#define sqlite3IsFixedLengthSerialType(t) ( (t)<12 || ((unsigned int)t)==SQLITE_MAX_U32 || ((unsigned int)t)==(SQLITE_MAX_U32-1) )
+#define sqlite3IsFixedLengthSerialType(t) ( (t)<12 || ((unsigned int)t)==SQLITE_MAX_U32 || ((unsigned int)t)==(SQLITE_MAX_U32-1) || ((unsigned int)t)==(SQLITE_MAX_U32-2) )
```

Comment:
I wonder maybe we should just change it to `( (t)<12 || ((unsigned int)t) >= (SQLITE_MAX_U32-2) )`.